Source/WebGPU/WebGPU/Buffer.h

```diff
@@ -148,6 +152,23 @@ class Buffer : public WGPUBufferImpl, public RefCounted<Buffer>, public CanMakeW
 #if CPU(X86_64)
     bool m_mappedAtCreation { false };
 #endif
+#if ENABLE(WEBGPU_SWIFT)
+} SWIFT_SHARED_REFERENCE(retainBuffer, releaseBuffer);
+#else
 };
+#endif
 
 } // namespace WebGPU
+
+#if ENABLE(WEBGPU_SWIFT)
+inline void retainBuffer(WebGPU::Buffer* obj)
```

Comment:
FYI: Swift-C++ has a bug where the very first refcount on a WebKit object will be wrong. So, for the time being, you need to avoid allocating refcounted objects from Swift. I believe the m_adoptionIsRequired ASSERT inside ref() is already sufficient to catch this bug in debug builds. So we don't need to do anything special yet.

Reply:
thanks for mentioning that!

Reply:
Maybe I don't have all the context, but do we need these `SWIFT_SHARED_REFERENCE` macros if we shouldn't be allocating refcounted objects from Swift?

I think I am missing some context as I am not sure where they are used.